### ISSUE TITLE
No module named '_rinterface_cffi_abi'

### ISSUE THREAD

**Issue by trvinh** [rpy2/rpy2#739]
**Describe the issue or bug**

Hi,
I am using rpy2 for one of my python package called `fcatpy`. It works fine if I directly run the python script with its full path (e.g.: `python calcCutoff.py -h`). But when I install the `fcatpy` package and call the function performed by that script, it shows me this error message:

```
(base) vinh@vinh-desktop:~/bionf/fCATpy$ fcat.cutoff -h
Traceback (most recent call last):
  File "/home/vinh/.local/lib/python3.8/site-packages/rpy2/rinterface_lib/openrlib.py", line 14, in <module>
    import _rinterface_cffi_api as _rinterface_cffi
ModuleNotFoundError: No module named '_rinterface_cffi_api'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vinh/.local/bin/fcat.cutoff", line 5, in <module>
    from fcatpy.calcCutoff import main
  File "/home/vinh/.local/lib/python3.8/site-packages/fcatpy/calcCutoff.py", line 30, in <module>
    from rpy2.robjects import FloatVector
  File "/home/vinh/.local/lib/python3.8/site-packages/rpy2/robjects/__init__.py", line 16, in <module>
    import rpy2.rinterface as rinterface
  File "/home/vinh/.local/lib/python3.8/site-packages/rpy2/rinterface.py", line 6, in <module>
    from rpy2.rinterface_lib import openrlib
  File "/home/vinh/.local/lib/python3.8/site-packages/rpy2/rinterface_lib/openrlib.py", line 16, in <module>
    import _rinterface_cffi_abi as _rinterface_cffi
ModuleNotFoundError: No module named '_rinterface_cffi_abi'
```
I think the import of FloatVector and importr somehow were not successful and gave those errors: 
```
from rpy2.robjects import FloatVector
from rpy2.robjects.packages import importr
```
However, it works by directly running the python script, which indicates that rpy2 was successfully installed and can be used. 
Many thanks for your help!

**To Reproduce**
```
pip install git+https://github.com/BIONF/fCATpy
fcat.cutoff -h
```

**Expected behavior**
The module '_rinterface_cffi_api' should be found while running the package function as same as directly running the python script.

**Comment by lgautier** [comment]
What does running the following in a terminal return ?
```bash
python -m rpy2.situation
```

**Comment by trvinh** [comment]
Hi @lgautier,
here is the output of that command:
```
(base) vinh@vinh-desktop:~$ python -m rpy2.situation
rpy2 version:
3.3.6
Python version:
3.7.6 (default, Jan  8 2020, 19:59:22) 
[GCC 7.3.0]
Looking for R's HOME:
    Environment variable R_HOME: None
    Calling `R RHOME`: /usr/lib/R
    Environment variable R_LIBS_USER: None
R's additions to LD_LIBRARY_PATH:
/usr/lib/R/lib:/usr/lib/x86_64-linux-gnu:/usr/lib/jvm/default-java/lib/server
R version:
    In the PATH: R version 4.0.2 (2020-06-22) -- "Taking Off Again"
    Loading R library from rpy2: OK
Additional directories to load R packages from:
None
C extension compilation:
  include:
  ['/usr/share/R/include']
  libraries:
  ['R', 'pcre2-8', 'lzma', 'bz2', 'z', 'rt', 'dl', 'm', 'icuuc', 'icui18n']
  library_dirs:
  ['/usr/lib/R/lib']
  extra_compile_args:
  []
  extra_link_args:
  ['-Wl,--export-dynamic', '-fopenmp', '-Wl,-Bsymbolic-functions', '-Wl,-z,relro']
```

**Comment by lgautier** [comment]
The fact that `python yourscript.py` works points out to a difference in environment between the two. For example, is a virtualenv used in both cases?

**Comment by trvinh** [comment]
I don’t think so. I tested both methods on the base env of anaconda. But I think it has something to do with the paths, maybe some conflicts between the python/R library paths from the system and anaconda. I am able to use both of the methods on other machines. I can’t, however, find a solution by comparing the paths between 2 computers :-(

**Comment by lgautier** [comment]
Anaconda seems to be at the root of quite a few similar issues with `rpy2`, and I am not using anaconda myself. Did you try contacting the maintainer of the R or rpy2 packages for anaconda?

**Comment by trvinh** [comment]
@lgautier no, not yet. But I have installed `rpy2` using `pip`, not with `conda install -c r rpy2`, so I thought asking directly here should be correct. Btw, I can use my package now after deleting anaconda and installing it again. It took quite some time to install all other packages, but it solved my issue here.
Many thanks for your time and help :-)

**Comment by zznx** [comment]
@lgautier 
here is the output of that command:
```
rpy2 version:
3.4.3
Python version:
3.6.12 (default, Oct 23 2020, 13:36:31) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
Looking for R's HOME:
    Environment variable R_HOME: None
    Calling `R RHOME`: /opt/R/4.0.0/lib/R
    Environment variable R_LIBS_USER: None
R's additions to LD_LIBRARY_PATH:
/path/to/R
R version:
    In the PATH: R version 4.0.0 (2020-04-24) -- "Arbor Day"
    Loading R library from rpy2: *** Error while loading: No module named '_rinterface_cffi_abi' ***
Additional directories to load R packages from:
None
C extension compilation:
  include:
  ['/opt/R/4.0.0/lib/R/include']
  libraries:
  ['R', 'pcre2-8', 'lzma', 'bz2', 'z', 'rt', 'dl', 'm', 'icuuc', 'icui18n']
  library_dirs:
  ['/usr/local/lib', '/opt/R/4.0.0/lib/R/lib']
  extra_compile_args:
  []
  extra_link_args:
  ['-Wl,--export-dynamic', '-fopenmp']
```

**Comment by HelloWorldLTY** [comment]
Hi all, I also have a problem related to this part, but my warnning is like I lost certrain files:

ImportError: libffi.so.7: cannot open shared object file: No such file or directory

My solution is conda install cffi, because libffi is based on cffi. Hope it can help!

**Comment by MakikoJuan** [comment]
Hi I have the same issue as you.  I already tried to do conda install cffi, but it still doesn't work.

**Comment by cpowr** [comment]
I had this problem as well, as I had copied a running version of rpy2 from base over to a virtual environment, but apparently without some crucial dependencies. I found the solution online on a [forum](https://forum.iolite.xyz/d/30-using-isoplotr-in-iolite/15) which hinted at all the missing dependencies.

In summary, in addition to copying the rpy2 folder, I had to copy the _cffi_backend.cp39-win_amd64.pyd and the cffi folder as well as some others. I believe this could also be resolved by uninstalling cffi and reinstalling it as suggested [here](https://stackoverflow.com/questions/34370962/no-module-named-cffi-backend).
